internal/provider/deployment.go

```diff
@@ -131,7 +131,7 @@ func (d *DeploymentDeployer) assignNodesHostIDs() error {
 			continue
 		}
 
-		for Contains(usedHosts, curHostID) {
+		for workloads.Contains(usedHosts, curHostID) {
```

Comment:
I like to have the contains on the workloads, except now it's misleading.

It should be `workloads.HasHostReserved(hostID)` and that implies that the workloads can tell the used hosts e.g `.ReservedHosts` or `.UsedHosts`, if that is possible, but if not I believe Contain as a function was fine.